### PR TITLE
include 3.8.0rc1

### LINKED ISSUE #313: Python 3.8
Issue to track adding Python 3.8 to the images, which it looks like should happen at the end of July, with 3.8.0b3.

From the [release notes for 3.8b1](https://www.python.org/downloads/release/python-380b1/), 3.8 ABI is planned to be frozen in beta 3, which is [scheduled](https://www.python.org/dev/peps/pep-0569/) for 2019-07-29.

Or maybe the policy in this repo should be freezing the ABI with release candidates? That would give a much shorter window for developers to start publishing wheels with this tool (3 weeks vs 3 months).

### PR DESCRIPTION
**Pull request by mattip**
Try to use ~3.8.0b4~ 3.8.0rc1 in the manylinux builds
closes #314
closes #313 
replaces #333

### CODE REVIEW

**Comment by mattip** [comment]
Once again the build fails
```
+ curl -fsSLO https://www.python.org/ftp/python/3.8.0/Python-3.8.0b4.tgz
+ curl -fsSLO https://www.python.org/ftp/python/3.8.0/Python-3.8.0b4.tgz.asc
+ gpg --verify Python-3.8.0b4.tgz.asc
gpg: Signature made Thu 29 Aug 2019 10:43:07 PM UTC using RSA key ID 10250568
gpg: Can't check signature: No public key
```

**Comment by lkollar** [comment]
Link to the upstream issue: https://bugs.python.org/issue37967.

**Comment by trishankatdatadog** [comment]
Automatic key download is a bad idea. It basically means: "trust whatever public key signed this package."

**Comment by mattip** [comment]
@trishankatdatadog thanks. I am not so good with gpg. Is the other solution the correct one? Where does the `fingerprint` value come from?
```
gpg --fetch-keys "https://keybase.io/ambv/pgp_keys.asc?fingerprint=e3ff2839c048b25c084debe9b26995e310250568"
```


Context for those not following the cpython issue

<details>
```
It looks like you don't have Łukasz key and your GnuPG is not configured for automatic key download.

Automatic key download works for me:

$ gpg --verify Python-3.8.0b4.tgz.asc 
gpg: assuming signed data in 'Python-3.8.0b4.tgz'
gpg: Signature made 2019-08-30T00:43:07 CEST
gpg:                using RSA key E3FF2839C048B25C084DEBE9B26995E310250568
gpg: requesting key 0xB26995E310250568 from hkp server keys.fedoraproject.org
gpg: key 0xB26995E310250568: public key "Łukasz Langa (GPG langa.pl) <lukasz@langa.pl>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: Good signature from "Łukasz Langa (GPG langa.pl) <lukasz@langa.pl>" [undefined]
gpg:                 aka "Łukasz Langa <lukasz@python.org>" [unknown]
gpg:                 aka "Łukasz Langa (Work e-mail account) <ambv@fb.com>" [undefined]
gpg:                 aka "[jpeg image of size 24479]" [unknown]


You could also download the key from keybase:

$ gpg --fetch-keys "https://keybase.io/ambv/pgp_keys.asc?fingerprint=e3ff2839c048b25c084debe9b26995e310250568"
gpg: requesting key from 'https://keybase.io/ambv/pgp_keys.asc?fingerprint=e3ff2839c048b25c084debe9b26995e310250568'
gpg: key 0xB26995E310250568: "Łukasz Langa (GPG langa.pl) <lukasz@langa.pl>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```

**Comment by trishankatdatadog** [comment]
@mattip That may be fine as a temporary workaround, but I recommend that [the official Python GPG public key list](https://www.python.org/static/files/pubkeys.txt) be updated before the final release of 3.8.0.

**Comment by matthew-brett** [comment]
Sorry to ask, but what is the current status of 3.8 builds for Manylinux?  I think the first release candidate is planned for today (30 September)?  https://www.python.org/dev/peps/pep-0569/#schedule

**Comment by mattip** [comment]
The gpg verify stage is still failing. python.org no longer recommends using the pubkeys.txt file, and now lists the fingerprints of the release manager on the [download page](https://www.python.org/downloads), and states "You can import a person's public keys from a public keyserver network server you trust by running a command like: `gpg --recv-keys [key id]` but (sorry for being such a gpg idiot) I don't understand how to get from the 3.8 release manager's fingerprint to the `key id`.

**Comment by matthew-brett** [comment]
I just did some futzing around.  I _think_ the idea is you check https://www.python.org/downloads/ - find the key id of the person, say Steve Dower, with key id "FC62 4643 4870 34E5" and then, assuming that https has protected you from deceit, do:

```
gpg --recv-keys "FC624643487034E5"
```

Is that what you meant?

**Comment by mattip** [comment]
@matthew-brett added that stanza in 4be5453, let's see if it works.

**Comment by mattip** [comment]
The build succeeds. It would be nice to be able to try out some 3.8 builds before the final release.

File: docker/build_scripts/build_env.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # source me
 
 PYTHON_DOWNLOAD_URL=https://www.python.org/ftp/python
-CPYTHON_VERSIONS="2.7.16 3.4.10 3.5.7 3.6.9 3.7.4"
+CPYTHON_VERSIONS="2.7.16 3.4.10 3.5.7 3.6.9 3.7.4 3.8.0b4"
```
Comment (auvipy) [inline docker/build_scripts/build_env.sh]: 3.8.0rc1 is now released

**Comment by matthew-brett** [comment]
Yes please to having the RC1 for testing wheel builds.

File: docker/build_scripts/build_env.sh (hunk 1 above)
Comment (mattip) [reply docker/build_scripts/build_env.sh]: updated and changed the PR name appropriately

**Pull request merged**